### PR TITLE
Plugin Management: make the base components in plugin management more generic.

### PR DESCRIPTION
**Pull request by yashwin**
#### Proposed Changes

This PR changes how the data is rendered mainly in the table and card view and makes it more generic so that other components like plugin details can use this. 

#### Testing Instructions

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout update/make-plugin-management-components-generic` and `yarn start-jetpack-cloud-p`
2. Open http://jetpack.cloud.localhost:3001/, and you'll be redirected to the /dashboard.
3. Click on **Plugins** in the sidebar -> Verify everything looks as expected

Screenshot 2022-08-03 at 3 21 01 PM

Screenshot 2022-08-10 at 3 22 34 PM

4. Click on **Switch Site** -> Select any site -> Click on **Plugins** -> Verify the plugins are shown as shown below. 

Screenshot 2022-08-04 at 12 31 56 PM

Screenshot 2022-08-11 at 2 52 52 PM

> **_NOTE:_** No design changes or logic changes have been done; only the way the components are rendered is changed.

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

The PR author and reviewer are responsible for completing the checklist.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes? Will be added later - 1202518759611394-as-1202627702018877
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React, or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202518759611394-as-1202829316329638

### CODE REVIEW

File: client/my-sites/plugins/plugin-management-v2/plugins-list/index.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+					) : (
+						rest.items.map( ( item ) => (
+							<PluginCard
+								{ ...rest }
+								key={ item[ primaryKey ] }
+								item={ item }
+								hasMoreActions={ hasMoreActions }
+							/>
+						) )
+					) }
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/plugins-list/index.tsx]: If we are consuming the properties in the component, IMO it would be better to specify these properties in the `PluginsList` declaration explicitly. Instead of `rest.title`, let's declare `title` and use it. We can do the same for other `rest.*` properties as well.

File: client/my-sites/plugins/plugin-management-v2/index.tsx [hunk 2]
```diff
@@ -64,30 +65,23 @@ export default function PluginManagementV2( {
 		return <div className="plugin-management-v2__no-sites">{ emptyStateMessage }</div>;
 	}
 
+	const title = translate( 'Installed Plugins' );
+
+	const rowFormatter = ( props: PluginRowFormatterArgs ) => {
+		return <PluginRowFormatter { ...props } selectedSite={ selectedSite } />;
+	};
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/index.tsx]: I can see we have the `rowFormatter` declared here, and then we just pass it down to the `PluginsList` component. Can we move that function to the `PluginsList` component itself to avoid passing it here? If that would require passing any extra props, it's okay to leave it as-is.

File: client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx [hunk 3]
```diff
@@ -1,78 +1,113 @@
...
+import type { PluginColumns, PluginRowFormatterArgs } from '../types';
 import type { SiteData } from 'calypso/state/ui/selectors/site-data';
-import type { ReactElement } from 'react';
+import type { ReactElement, ReactNode } from 'react';
 
 import './style.scss';
 
 interface Props {
-	item: Plugin;
+	item: any;
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx]: Can we use the type `Plugin` here? Also, what do you think about renaming the `item` to `plugin`? I think that describes better the nature of the expected items.

File: client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx [hunk 4]
```diff
@@ -1,78 +1,113 @@
...
+	rowFormatter,
+	columns,
+	hasMoreActions,
+}: Props ): ReactElement {
+	const columnKeys: { [ key: string ]: boolean } = columns.reduce(
+		( obj, cur ) => ( { ...obj, [ cur.key ]: true } ),
+		{}
+	);
+
+	const showLeftContent = columnKeys[ 'plugin' ];
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx]: When we don't want to display this left content HTML? Is that something related to the plugin details UI?

File: client/my-sites/plugins/plugin-management-v2/plugins-list/index.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import PluginsTable from '../plugins-table';
+import type { PluginColumns, PluginRowFormatterArgs } from '../types';
+import type { SiteData } from 'calypso/state/ui/selectors/site-data';
+import type { ReactElement, ReactNode } from 'react';
+
+import './style.scss';
+
+interface Props {
+	selectedSite: SiteData;
+	items: Array< any >;
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/plugins-list/index.tsx]: Can we use `Array< Plugin >` here? And what do you think about renaming `items` to `plugins` here as well?

File: client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx (hunk 4 above)
Comment (yashwin) [reply client/my-sites/plugins/plugin-management-v2/plugin-card/index.tsx]: Yes, something which will be needed.

**Review by vitozev — APPROVED** [review]
Thanks for making these changes! LGTM!

**Comment by yashwin** [comment]
> Thanks for making these changes! LGTM!

Thanks for your review and feedback.
